src/python/nimbusml/pipeline.py

```diff
@@ -819,6 +819,12 @@ def _fit_graph(self, X, y, verbose, **params):
                     model=output_predictor_model)
                 output_predictor_model_node._implicit = True
                 graph_nodes.append(output_predictor_model_node)
+        elif learner_node == None and transform_models[-1].startswith("$dataset_transformer_model"):
+            single_model_node = transforms_modelcombiner(
+                models=transform_models,
+                output_model=output_model)
+            single_model_node._implicit = True
+            graph_nodes.append(single_model_node)
```

Comment:
It looks like the body of the `if` statement matches the body of the `if` statement below it. Could the condition in the `if` statement be moved to the one below it?

```python
elif len(transform_models) > 1 or \
    transform_models[-1].startswith("$dataset_transformer_model")
```

This would reduce the amount of duplicate code.

Also, would there be an array index error if `transform_models` was empty?

Reply:
We can reduce the amount of duplicate code and ensure no index error can happen with this snippet:
```
        elif len(transform_models) > 1 or \
            (learner_node == None and len(transform_models) > 0 and \
            transform_models[-1].startswith("$dataset_transformer_model")):
```